src/jdk.jlink/share/classes/jdk/tools/jlink/internal/Platform.java

```diff
@@ -24,12 +24,42 @@
  */
 package jdk.tools.jlink.internal;
 
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteOrder;
+import java.util.HashMap;
 import java.util.Locale;
+import java.util.Map;
+import java.util.Properties;
 
 /**
  * Supported platforms
  */
-public record Platform(OperatingSystem os, Architecture arch) {
+public record Platform(OperatingSystem os, Architecture arch, ByteOrder endianness) {
+    private static final Properties KNOWN_ENDIANNESS;
+    private static final String ENDIANNESS_KEY_SUFFIX = ".endianness";
+
+    static {
+        Properties p = null;
+        try (InputStream is = Platform.class.getResourceAsStream("target.properties")) {
+            p = new Properties();
+            p.load(is);
+        } catch (IOException e) {
+            throw new ExceptionInInitializerError(e);
+        }
+        Properties endianness = new Properties();
+        for (String key : p.stringPropertyNames()) {
+            if (key.endsWith(ENDIANNESS_KEY_SUFFIX)) {
+                String val = p.getProperty(key);
+                String platform = key.substring(0, key.indexOf(ENDIANNESS_KEY_SUFFIX));
+                if (platform.isEmpty()) {
+                    throw new InternalError("Incorrect key '" + key + "'");
+                }
```

Comment:
If you scan all entries, this can probably check the value is valid (`little` or `big`) and store them in a `Map<String, ByteOrder>` so that `parsePlatform` can simply get the byte order by `KNOWN_ENDIANNESS.get(platformString)`.

Or the other option can just be doing in the `parsePlatform` method and not needed to scan all entries.

```
   String v = KNOWN_ENDIANNESS.getProperty(platformString + ENDIANNESS_KEY_SUFFIX);
```

Reply:
Hello Mandy, I was in two minds when I added that iteration logic. I wasn't sure if that was necessary or we could just check for the values when looking up. I have now updated the PR to remove the iteration and just rely on looking up the key, as you suggested, and then checking the value.